lib/index.js

```diff
@@ -89,44 +88,81 @@ AdWords.prototype.page = function(page) {
  */
 
 AdWords.prototype.track = function(track) {
-  var id = this.options.conversionId;
+  var self = this;
   var props = track.properties();
-  var remarketing = this.options.remarketing;
-  var events = this.events(track.event());
+  var eventMappings = this.options.eventMappings;
   var revenue = track.revenue() || 0;
-  // Check if this is a whitelisted event for standalone remarketing tag
-  var whitelisted = this.options.whitelist.indexOf(track.event()) > -1;
-  var sentAlready = false;
 
-  each(function(label) {
-    delete props.revenue;
-    // Fire conversion tag
-    window.google_trackConversion({
-      google_conversion_id: id,
-      google_custom_params: props,
-      google_conversion_language: 'en',
-      google_conversion_format: '3',
-      google_conversion_color: 'ffffff',
-      google_conversion_label: label,
-      google_conversion_value: revenue,
-      google_remarketing_only: false // ensure this is a conversion tag
+  if (eventMappings) { // if eventMappings exists then the new metadata schema is being used
+    eventMappings.forEach(function(mapping) {
+      if (mapping.value.eventName.toLowerCase() !== track.event().toLowerCase()) return;
+      var id = mapping.value.conversionId === '' ?  self.options.conversionId : mapping.value.conversionId;  // customer can either specify one global conversion id or one per event<->label mapping
+
+      // Fire conversion tag
+      if (mapping.value.label !== '') {
+      delete props.revenue;
+      
+        window.google_trackConversion({
+          google_conversion_id: id,
+          google_custom_params: props,
+          google_conversion_language: 'en',
+          google_conversion_format: '3',
+          google_conversion_color: 'ffffff',
+          google_conversion_label: mapping.value.label,
+          google_conversion_value: revenue,
+          google_remarketing_only: false // ensure this is a conversion tag
+        });
+      }
+
+      // Fire remarketing tag
+      if (mapping.value.remarketing) {
+        window.google_trackConversion({
+          google_conversion_id: id,
+          google_custom_params: props, // do not send PII here!
+          google_remarketing_only: true // ensure this is a remarketing tag
+        });
+      }
     });
-    // Fire remarketing tag
-    if (!sentAlready && remarketing) {
+  } 
+  // DELETE BELOW AND TAKE ABOVE OUT OF IF CLAUSE, only exists for metadata cutover
+  else {  // eslint-disable-line
+    var id = this.options.conversionId;
+    var events = this.events(track.event());
+    // Check if this is a whitelisted event for standalone remarketing tag
+    var whitelisted = this.options.whitelist.indexOf(track.event()) > -1;
+    var sentAlready = false;
+    var remarketing = this.options.remarketing;
+  
+    each(function(label) {
+      delete props.revenue;
```

Comment:
We can probably bump this up to where we set the vars - no need to delete props.revenue for every label, since it'll be gone after the first round.

Reply:
This is an exact copy of the legacy code and it's going to be deleted after this metadata cutover's been completed so I'd prefer just not to touch it

Reply:
Sounds fair.